pip/req/req_set.py

```diff
@@ -253,6 +253,10 @@ def add_requirement(self, install_req, parent_req_name=None):
                     # scanning.
                     result = []
                 elif not install_req.constraint:
+                    if install_req.editable:
```

Comment:
I think perhaps this can be if install_req.url is not None or something like that. That will signal we have a path rather than a version.